examples/mongocxx/causal_consistency.cpp

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <ctime>
+#include <iostream>
+
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/builder/stream/helpers.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/client_session.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/instance.hpp>
+#include <mongocxx/options/client_session.hpp>
+#include <mongocxx/options/insert.hpp>
+#include <mongocxx/read_concern.hpp>
+#include <mongocxx/uri.hpp>
+#include <mongocxx/write_concern.hpp>
+
+using bsoncxx::builder::stream::document;
+using bsoncxx::builder::stream::open_document;
+using bsoncxx::builder::stream::close_document;
+using bsoncxx::builder::stream::finalize;
+
+int main() {
+    using namespace mongocxx;
+
+    instance inst{};
+    client client{mongocxx::uri{"mongodb://localhost/?replicaSet=replset"}};
+
+    // Start Causal Consistency Example 1
+
+    write_concern wc_majority{};
+    wc_majority.acknowledge_level(write_concern::level::k_majority);
```

Comment:
You can remove this line, setting `majority` below also sets the acknowledge level.

Reply:
Ah of course! Done.